architecture/trusted_applications.rst

```diff
@@ -76,13 +76,25 @@ the suffix ``.ta``. They are built separately from the OP-TEE core boot-time
 blob, although when they are built they use the same build system, and are
 signed with the key from the build of the original OP-TEE core blob.
 
-Because the TAs are signed, they are able to be stored in the untrusted REE
-filesystem, and ``tee-supplicant`` will take care of passing them to be checked
-and loaded by the Secure World OP-TEE core. Note that this type of TA isn't
-encrypted.
+The signed TA can optionally be encrypted in case TA build option:
+``CFG_ENCRYPT_TA=y`` is set and symmetric key is provided via TA build option:
+``TA_ENC_KEY=<hex string representing 256 bit key>``. The overall signature and
+encryption scheme follows ``Sign-then-Encrypt-then-MAC``. During OP-TEE
+runtime, the symmetric key used to decrypt TA has to be provided in a platform
+specific manner via overriding API:
 
-REE filesystem TAs come in two formats, the legacy TA and the bootstrap TA.
-The bootstrap TA format is used by ``scripts/sign.py`` since version 3.7.0.
+.. code-block:: c
+
+    TEE_Result tee_otp_get_ta_enc_key(uint32_t key_type, uint8_t *buffer,
+                                      size_t len);
+
+Because the TAs are signed (optionally encrypted), they are able to be stored
```

Comment:
This paragraph can move at the beginning of the section, in the introduction before mentioning the three types of TAs.

Reply:
Okay.